fgpyo/sam/__init__.py

```diff
@@ -180,8 +180,16 @@
 from pysam import AlignmentFile as SamFile
 from pysam import AlignmentHeader as SamHeader
 from pysam import qualitystring_to_array
-from typing_extensions import Self
-from typing_extensions import deprecated
+
+if sys.version_info > (3, 10):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+if sys.version_info > (3, 12):
```

Comment:
Added in 3.13: https://docs.python.org/3/library/warnings.html#warnings.deprecated

Reply:
You should do this instead so you are not affected by the `micro` field of the version.

```suggestion
if sys.version_info[:2] >= (3, 13):
```